src/routes/measurement-threshold/find.ts

```diff
@@ -0,0 +1,110 @@
+import Joi from 'joi';
+import { Edge } from '../../models/edge';
+import { Pin } from '../../models/pin';
+import { ControllerGeneratorOptionsWithClient } from '../../comms/controller';
+
+import { PinGroup } from '../../models/pin-group';
+
+import { TableQuery, EffectiveTableQuery, tableQuerySchemaGenerator } from '../../comms/table-controller';
+import { Quantity } from '../../models/quantity';
+import { stringBeforeV7ElseStringOrTranslationSchema } from '../../models/string-or-translations';
+import { schema as siNumberSchema, SiNumber } from '../../models/si-number';
+
+interface Query extends TableQuery {
+  quantityHashId?: string | null;
+  pinGroupHashId?: string | null;
+  edgeHashId?: string | null;
+  pinHashId?: string | null;
+  type?: 'quantity' | 'port';
+}
+
+type Request = {
+  query?: Query;
+} | undefined;
+
+interface EffectiveQuery extends EffectiveTableQuery {
+    quantityHashId?: string | null;
+    pinGroupHashId?: string | null;
+    edgeHashId?: string | null;
+    pinHashId?: string | null;
+    type: 'quantity' | 'port';
+}
+
+interface EffectiveRequest {
+  query: EffectiveQuery;
+}
+
+interface Threshold {
+  criticallyLow: SiNumber | null;
+  low: SiNumber | null;
+  high: SiNumber | null;
+  criticallyHigh: SiNumber | null;
+}
+
+interface ResponseRow {
+  quantity: Pick<Quantity, 'name' | 'hashId'>;
+  location: {
+    pinGroup: Pick<PinGroup, 'name' | 'hashId'>;
+    pin: Pick<Pin, 'name' | 'hashId'>;
+    edge: Pick<Edge, 'name' | 'hashId'> | null;
+    } | null;
+  threshold: Threshold;
+}
+
+interface Response {
+  nextPageOffset: string | null;
+  rows: ResponseRow[];
+}
+
+const controllerGeneratorOptions: ControllerGeneratorOptionsWithClient = {
+  method: 'get',
+  path: '/',
+  query: tableQuerySchemaGenerator(Joi.string().valid('quantityName', 'pinGroupName', 'hashId').default('quantityName')).keys({
+    quantityHashId: Joi.string().allow(null).default(null),
+    pinGroupHashId: Joi.string().allow(null).default(null),
+    edgeHashId: Joi.string().allow(null).default(null),
+    pinHashId: Joi.string().allow(null).default(null),
+    type: Joi.string().valid('quantity', 'port'),
+  }),
+  right: { environment: 'THRESHOLDS' },
+  response: (apiVersion: number): Joi.ObjectSchema => Joi.object().keys({
+    nextPageOffset: Joi.string().allow(null).example(null).required()
+      .description('This is the last page if nextPageOffset is null'),
+    rows: Joi.array().items(Joi.object().keys({
+      quantity: Joi.object().keys({
+        name: stringBeforeV7ElseStringOrTranslationSchema(apiVersion).required().example('Temperature'),
```

Comment:
as this route does not exist before v7, I think we can use [`stringOrTranslationSchema`](https://github.com/withthegrid/platform-sdk/blob/a61dfba3662c7a92e6e87257e1c6176b439baa2e/src/models/string-or-translations.ts#L5-L8) here?

Reply:
Thanks, changing